sdks/java/extensions/euphoria/euphoria-core/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/client/operator/base/Builders.java

```diff
@@ -81,11 +119,45 @@
      * @param <W> the type of the windowing
      * @param windowing the windowing strategy to apply to the input dataset
      * @return the next builder to complete the setup of the {@link ReduceByKey} operator
+     * @deprecated This method is deprecated and will be removed in future. It is left here for
+     * backward compatibility.
      */
-    <W extends Window<W>> Object windowBy(Windowing<InputT, W> windowing);
+    @Deprecated
+    <W extends Window<W>> Object windowBy(Windowing<?, W> windowing);
+  }
+
+  /**
+   * Second builder in windowing builders chain. It introduces a {@link Trigger}.
+   *
+   * @param <OutAccBuilderT> following {@link AccumulatorMode} builder type
+   */
+  public interface TriggeredBy<OutAccBuilderT extends AccumulatorMode> {
+
+    OutAccBuilderT triggeredBy(@Nonnull Trigger trigger);
   }
 
-  /** TODO: complete javadoc. */
+  /**
+   * Third and last builder in windowing chain introducing {@link WindowingStrategy.AccumulationMode
+   * accumulation mode}.
+   *
+   * @param <OutBuilderT> output builder type
+   */
+  public interface AccumulatorMode<OutBuilderT> {
+
+    OutBuilderT accumulationMode(@Nonnull WindowingStrategy.AccumulationMode accumulationMode);
```

Comment:
All params should be nonnull, unless annotated otherwise.

Reply:
Removed.